### ISSUE TITLE
Unable to sync with OwnCloud7

### ISSUE THREAD

**Issue by skarra** [skarra/ASynK#49]
On master @ 4c602a4d948a68459e002fecd5eedb83c4906c88 having trouble synching with oc7. Others, in particular @rrix are able to sync; I am not sure how. This is a possible bug in ASynK, but this works quite well with Apple CalendarServer, Baikal, and other such servers.

Logs are as below:

```
Srirams-MacBook-Pro:asynk.co sriramkarra$ python ./asynk.py --op=sync --user-dir=~/.asynk.cd --name=oc7 --dry-run --log=debug
Applying base config from file /Users/sriramkarra/stuff/code/python/asynk.co/config/config_v7.json...
Applying base config from file /Users/sriramkarra/stuff/code/python/asynk.co/config/config_v7.json...done
Applying user customizations from file /Users/sriramkarra/.asynk.cd/config.py...
Applying user customizations from file /Users/sriramkarra/.asynk.cd/config.py...done
[10:52:35.328     INFO] Debug logging to file: /Users/sriramkarra/.asynk.cd/logs/asynk_logs.2014-09-28.10-52-35.327743
[10:52:35.332     INFO] Parsing BBDB file /Users/sriramkarra/dotfiles/.bbdb.misc/bbdb.testbbcd.own...
[10:52:35.334    DEBUG] New BBContactsFolder: default
[10:52:35.334     INFO] Parsing BBDB Store with encoding utf-8...
[10:52:35.358     INFO] Parsing BBDB Store with encoding utf-8...Success
[10:52:35.359     INFO] Successfully parsed 9 entries.
[10:52:35.454     INFO] Logging into CardDAV server: http://cloud.fort.kickass.systems:10080/owncloud/remote.php/carddav/principals/skarra/ - user: skarra...
[10:52:35.455    DEBUG] PIMDB cd does not have email_domains.
[10:52:35.455    DEBUG] PIMDB cd does not have postal_map
[10:52:35.455    DEBUG] PIMDB cd does not have notes_map
[10:52:35.455    DEBUG] PIMDB cd does not have phones_map
[10:52:35.456    DEBUG] Sending request OPTIONS to server cloud.fort.kickass.systems:10080/owncloud/remote.php/carddav/principals/skarra/...
[10:52:36.733    DEBUG] Sending Request URL: /owncloud/remote.php/carddav/principals/skarra/	Method: OPTIONS
[10:52:37.335    DEBUG] Received Response: HTTP/1.1 401 Unauthorized

[10:52:37.335    DEBUG] Sending request OPTIONS to server cloud.fort.kickass.systems:10080/owncloud/remote.php/carddav/principals/skarra/...
[10:52:37.335    DEBUG] Sending Request URL: /owncloud/remote.php/carddav/principals/skarra/	Method: OPTIONS
[10:52:37.858    DEBUG] Received Response: HTTP/1.1 200 OK

[10:52:37.858    DEBUG] Sending Request URL: /owncloud/remote.php/carddav/principals/skarra/	Method: PROPFIND
[10:52:38.492    DEBUG] Received Response: HTTP/1.1 207 Multi-Status

Found current principal path: /owncloud/remote.php/carddav/principals/skarra/
[10:52:38.784    DEBUG] Sending Request URL: /owncloud/remote.php/carddav/principals/skarra/	Method: PROPFIND
[10:52:39.405    DEBUG] Received Response: HTTP/1.1 207 Multi-Status

[10:52:39.407    DEBUG] Found principal: skarra
[10:52:39.407    DEBUG] CDPIMDB.fetch_folders(): Begin
[10:52:39.407    DEBUG] Exploring following root paths: ['/owncloud/remote.php/carddav/addressbooks/skarra/']
[10:52:39.409    DEBUG] Processing root path /owncloud/remote.php/carddav/addressbooks/skarra/
[10:52:39.697    DEBUG] Sending Request URL: /owncloud/remote.php/carddav/addressbooks/skarra/	Method: PROPFIND
[10:52:40.293    DEBUG] Received Response: HTTP/1.1 207 Multi-Status

[10:52:40.293    DEBUG]   Found following properties: ['/owncloud/remote.php/carddav/addressbooks/skarra/'].
[10:52:40.293    DEBUG] CDPIMDB.fetch_folders(): Done.
[10:52:40.602    DEBUG] Sending Request URL: /owncloud/remote.php/carddav/addressbooks/skarra/	Method: PROPFIND
[10:52:41.221    DEBUG] Received Response: HTTP/1.1 207 Multi-Status

[10:52:41.222    DEBUG] Coud not find default adbk Property. Trying to use first of the available contacts folders
Traceback (most recent call last):
  File "./asynk.py", line 1125, in <module>
    main()
  File "./asynk.py", line 80, in main
    asynk.dispatch()
  File "./asynk.py", line 511, in dispatch
    res = getattr(self, self.get_op())()
  File "./asynk.py", line 775, in op_sync
    pname = self._load_profile()
  File "./asynk.py", line 1119, in _load_profile
    self._login()
  File "./asynk.py", line 312, in _login
    self.set_db(db2id, getattr(self, login_func)())
  File "./asynk.py", line 1061, in login_cd
    self.get_cduser(), self.get_cdpw())
  File "/Users/sriramkarra/stuff/code/python/asynk.co/asynk/pimdb_cd.py", line 57, in __init__
    self.set_def_folders()
  File "/Users/sriramkarra/stuff/code/python/asynk.co/asynk/pimdb_cd.py", line 135, in set_def_folders
    assert(len(fs) > 0)
AssertionError
```

**Comment by zvonko123** [comment]
https://github.com/skarra/ASynK/issues/74

**Comment by skarra** [comment]
Fixed via #74
